### PR TITLE
Add support for Zeitwerk

### PR DESCRIPTION
**Pull request by kennyadsl**
Decorator files are now correctly added to the autoloader and are also loaded on engine boot. This extension is quite unique since it conditionally loads frontend and backend so we can't use the Engine helpers recently added in `solidus_support`. 

This PR is also removing useless assets compilation and refactoring the Engine a little bit.

### CODE REVIEW

**Comment by aldesantis** [comment]
@kennyadsl how about moving decorators into their own directory (like we do in `app`) and having a structure like the following instead?

- `lib/controllers/backend`
- `lib/decorators/backend`
- ...

So, basically, the pattern is: `lib/[file type]/[solidus engine]`.

**Comment by kennyadsl** [comment]
We'd still need to differentiate the type of decorator in an inner folder, so it would be:

- `lib/controllers/backend`
- `lib/decorators/backend/controllers/...`

So it's a bit more complex, but maybe the best path.

**Comment by aldesantis** [comment]
I think that works, even if it's not exactly beautiful, and it's basically the same structure we have in `app/decorators` so it's also consistent.

File: lib/spree/auth/engine.rb [hunk 1]
```diff
@@ -20,22 +20,41 @@ class Engine < Rails::Engine
...
-          lib/assets/javascripts/spree/backend/solidus_auth.css
-        ]
+      def self.load_decorators_for(component_name)
+        base_path = root.join('lib/decorators', component_name)
+
+        if Rails.respond_to?(:autoloaders) && Rails.autoloaders.main
+          # Add decorators folder to the Rails autoloader. This
+          # allows Zeitwerk to resolve decorators paths correctly,
+          # when used.
+          Dir.glob(base_path.join('*')) do |decorators_folder|
```
Comment (elia) [inline lib/spree/auth/engine.rb]:
```suggestion
          base_path.glob('*') do |decorators_folder|
```

File: lib/spree/auth/engine.rb [hunk 2]
```diff
@@ -20,22 +20,41 @@ class Engine < Rails::Engine
...
+          end
+        end
 
-        Dir.glob(File.join(File.dirname(__FILE__), "../../controllers/backend/*/*/*_decorator*.rb")) do |c|
-          Rails.configuration.cache_classes ? require(c) : load(c)
+        # Load decorator files. This is needed since they are
+        # never explicitely referenced in the application code
+        # and won't be loaded by default. We need them to be
+        # executed anyway to extend exisiting classes.
+        Dir.glob(base_path.join('**/*_decorator*.rb')) do |decorator_path|
```
Comment (elia) [inline lib/spree/auth/engine.rb]:
```suggestion
        base_path.glob('**/*_decorator*.rb') do |decorator_path|
```

**Pull request merged**